### PR TITLE
Improve error message for local constant declaration

### LINKED ISSUE #29864: Unclear parser error message when `const` is used within a function
**Description:**

```ballerina
public function main() {
    const expected = "";
}
```

Out:
```
ERROR [main.bal:(31:1,31:1)] missing close brace token
ERROR [main.bal:(33:2,33:2)] invalid token '}'
```

The errors shown above are very unintuitive to a new user.

**Steps to reproduce:**

**Affected Versions:**
SLP alpha4

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

Fixes #29864

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties [hunk 1]
```diff
@@ -313,3 +313,4 @@ error.metadata.not.attached.to.a.object.member=metadata not attached to a object
 error.type.desc.after.rest.descriptor=type descriptor not allowed after rest descriptor
 error.configurable.var.implicitly.final=''final'' qualifier not allowed: configurable variables are implicitly final
 error.missing.attach.point.name=missing attach point name
+error.local.const.decl.not.allowed=local constant declaration not allowed
```
Comment (MaryamZi) [inline compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties:316]: How about something like
```suggestion
error.local.const.decl.not.allowed=constant declarations are allowed only at module level
```

IMO, saying "local constant declaration" kinda implies the existence of a "local constant declaration" construct. WDYT?

Comment (rdulmina) [reply compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties:316]: +1 will update

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java [hunk 2]
```diff
@@ -2782,9 +2782,7 @@ private STNode parseFunctionBodyBlock(boolean isAnonFunc) {
                 break;
             }
 
-            //Local type def is not allowed in new spec, hence add it as invalid node minutia.
-            if (stmt.kind == SyntaxKind.LOCAL_TYPE_DEFINITION_STATEMENT) {
-                addInvalidNodeToNextToken(stmt, DiagnosticErrorCode.ERROR_LOCAL_TYPE_DEFINITION_NOT_ALLOWED);
+            if (isStmtNotAllowedInLocalLevel(stmt)) {
```
Comment (lochana-chathura) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:2785]: the method name is bit misleading. It looks like just a boolean check, but it also invalidates the node. 
HBT checkAndInvalidateUnallowedStmts?

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java [hunk 3]
```diff
@@ -68,6 +68,7 @@ public void test(String fileName, String path) throws IOException {
                 "annotations_source_04.bal", // could be considered an invalid scenario
                 "receive_action_source_01.bal", // issue #26376
                 "doc_source_21.bal", // issue #28172
+                "module_var_decl_source_18.bal", // issue #31307
```
Comment (rdulmina) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java:71]: Issue #31307 to enable the test

**Review by lochana-chathura — APPROVED** [review]

**Review by KRVPerera — APPROVED** [review]

**Pull request merged**